### ISSUE TITLE
Open upstream Request For Enhancement to export a sentinel error (or constant) for `x509: certificate relies on legacy Common Name field, use SANs instead`

### ISSUE THREAD

**Issue by atc0005** [atc0005/check-cert#520]
## Overview

GH-276 added initial support for matching the `x509: certificate relies on legacy Common Name field, use SANs instead` `x509.HostnameError` error text and offering sysadmins advice for how to resolve it.

Later work in this project shifted how this detection/advice was handled, but the core implementation remains the same:

1. detect this error condition using string matching
2. surface the error with advice to the sysadmin

<details>
<summary>Show implementation</summary>

https://github.com/atc0005/check-cert/blob/aa19a9c4779df09b54de8bffb85adbe80df1e7f5/internal/certs/certs.go#L77-L83

https://github.com/atc0005/check-cert/blob/aa19a9c4779df09b54de8bffb85adbe80df1e7f5/internal/certs/validation-hostname.go#L167-L197

</details>

Ideally we would not have to perform string matching to detect the error, but due to how the error is implemented upstream I am not aware of any other option:

```go
// ...
func (h HostnameError) Error() string {
	c := h.Certificate

	if !c.hasSANExtension() && matchHostnames(c.Subject.CommonName, h.Host) {
		return "x509: certificate relies on legacy Common Name field, use SANs instead"
	}
// ...
```

I don't know what is required to open a proposal or Request for Enhancement (RFE) in the upstream Go project, but that appears to be the next step to have the error surfaced as an value that we can reliably match against. While string matching works now, I don't know how long it will continue to be provided with that exact pattern.

## References

- https://github.com/atc0005/check-cert/blob/aa19a9c4779df09b54de8bffb85adbe80df1e7f5/internal/certs/certs.go
- https://github.com/atc0005/check-cert/pull/358
- https://github.com/atc0005/check-cert/issues/276
- https://github.com/atc0005/check-cert/issues/70
- https://cs.opensource.google/go/go/+/refs/tags/go1.20.1:src/crypto/x509/verify.go;l=104
